### PR TITLE
Providing raw data as a CSV file

### PR DESCRIPTION
**Pull request by gtorodelvalle**
- 100% tests passed

### CODE REVIEW

File: src/sth_database.js [hunk 1]
```diff
@@ -342,6 +374,42 @@
...
+   * @param {object} stream The stream
+   * @param {string} fileName The file name where the stream contents should be stored
+   */
+  function save2File(attrName, stream, fileName) {
+    // Create the directory for the logging files if it does not exist
+    if (fs.existsSync(sthConfig.TEMPORAL_DIR)) {
+      // The path exists
+      if (!fs.statSync(sthConfig.TEMPORAL_DIR).isDirectory()) {
+        // The paths exists but it is not a directory
+        mkdirp.sync(sthConfig.TEMPORAL_DIR);
```
Comment (mrutid) [inline src/sth_database.js]: don't it do the same thing on every branch?  

mkdirp.sync(sthConfig.TEMPORAL_DIR);

**Comment by dmoranj** [comment]
I imagine is not part of this particular PR, but, please, avoid functions more than 100 lines in length. Code snipets like the following one (extracted from `sth_database.js`) should be avoided at all costs:
```
                  }
                );
              }
            }
          );
        }
      }
    };
```

File: src/sth_database.js [hunk 2]
```diff
@@ -355,7 +416,7 @@
    * @param {Function} callback Callback to inform about any possible error or results
    */
   function getRawData(collection, entityId, entityType, attrName, lastN, hLimit, hOffset,
```
Comment (dmoranj) [inline src/sth_database.js]: Cyclomatic complexity in this function seems huge at first sight. Please, split the function functionality in smaller functions, to improve readibility.

**Comment by dmoranj** [comment]
Please, tend to use smaller functions. Some of them (`getRawData()` in `sth_server.js` is a good example are really difficult to read.

File: src/sth_database.js [hunk 3]
```diff
@@ -342,6 +374,35 @@
   }
 
   /**
+   * Saves the contents of a readable stream into a file in the local file system
+   * @param {string} attrName The name of the attribute the stream contents refers to
+   * @param {object} stream The stream
+   * @param {string} fileName The file name where the stream contents should be stored
+   */
+  function save2File(attrName, stream, fileName) {
```
Comment (dmoranj) [inline src/sth_database.js]: These two functions are modelic in its scope and length (related to the comments below about function length).

File: src/sth_database.js (hunk 2 above)
Comment (gtorodelvalle) [inline src/sth_database.js]: Agreed and to be solved in a forthcoming PR including the refactoring of the whole app file structure to make it follow Telefónica R&D's Node.js application guidelines... ;) Thanks for pointing it out... ;)

**Comment by gtorodelvalle** [comment]
Agreed and to be solved in a forthcoming PR including the refactoring of the whole app file structure to make it follow Telefónica R&D's Node.js application guidelines... ;) Thanks for pointing it out... ;)

**Comment by anabelengp** [comment]
LGTM

**Pull request merged**